In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [2]:
dataset2018 = pd.read_json('cve-2018.json')

In [3]:
dataset2018.shape

(13949, 6)

In [4]:
dataset2018.columns.values

array(['CVE_Items', 'CVE_data_format', 'CVE_data_numberOfCVEs',
       'CVE_data_timestamp', 'CVE_data_type', 'CVE_data_version'],
      dtype=object)

In [5]:
dataset2018.CVE_Items[200]

{'configurations': {'CVE_data_version': '4.0',
  'nodes': [{'cpe': [{'cpe22Uri': 'cpe:/o:microsoft:windows_10:1511',
      'cpe23Uri': 'cpe:2.3:o:microsoft:windows_10:1511:*:*:*:*:*:*:*',
      'vulnerable': True},
     {'cpe22Uri': 'cpe:/o:microsoft:windows_10:1607',
      'cpe23Uri': 'cpe:2.3:o:microsoft:windows_10:1607:*:*:*:*:*:*:*',
      'vulnerable': True},
     {'cpe22Uri': 'cpe:/o:microsoft:windows_server_2016:-',
      'cpe23Uri': 'cpe:2.3:o:microsoft:windows_server_2016:-:*:*:*:*:*:*:*',
      'vulnerable': True}],
    'operator': 'OR'}]},
 'cve': {'CVE_data_meta': {'ASSIGNER': 'cve@mitre.org', 'ID': 'CVE-2017-0219'},
  'affects': {'vendor': {'vendor_data': [{'product': {'product_data': [{'product_name': 'windows_10',
         'version': {'version_data': [{'version_value': '1511'},
           {'version_value': '1607'}]}},
        {'product_name': 'windows_server_2016',
         'version': {'version_data': [{'version_value': '-'}]}}]},
      'vendor_name': 'microsoft'}]}},
  

In [6]:
dataset2018.CVE_Items[200].keys()

dict_keys(['cve', 'configurations', 'impact', 'publishedDate', 'lastModifiedDate'])

In [15]:
rev = []
novals = []
j=0
for i in range(dataset2018.shape[0]):
    new=dataset2018.CVE_Items[i]
    if('baseMetricV2' in new['impact'].keys()):
        var=new['impact']['baseMetricV2']['severity']
        rev.append(var)
    else:
        rev.append('Not Exists')
        novals.append(j)
    j=j+1  

In [10]:
rev=np.array(rev)

In [11]:
rev.shape

(13949,)

In [12]:
num = 0
val = []
for i in range(rev.shape[0]):
    if (rev[i] == 'Not Exists'):
        num = num + 1
        val.append(i)

In [13]:
print(num)

1817


### so out of 13949 rows , 1817 doesn't have severity value in there 

### an example of row that doesn't has severity in there 

In [17]:
novals=np.array(novals)

In [18]:
novals.shape

(1817,)

In [19]:
novals[0]

330

In [23]:
dataset2018.CVE_Items[331]

{'configurations': {'CVE_data_version': '4.0', 'nodes': []},
 'cve': {'CVE_data_meta': {'ASSIGNER': 'cve@mitre.org', 'ID': 'CVE-2017-0357'},
  'affects': {'vendor': {'vendor_data': []}},
  'data_format': 'MITRE',
  'data_type': 'CVE',
  'data_version': '4.0',
  'description': {'description_data': [{'lang': 'en',
     'value': 'A heap-overflow flaw exists in the -tr loader of iucode-tool starting with v1.4 and before v2.1.1, potentially leading to SIGSEGV, or heap corruption.'}]},
  'problemtype': {'problemtype_data': [{'description': []}]},
  'references': {'reference_data': [{'url': 'http://www.securityfocus.com/bid/95432'},
    {'url': 'https://gitlab.com/iucode-tool/iucode-tool/issues/3'},
    {'url': 'https://security-tracker.debian.org/tracker/CVE-2017-0357'}]}},
 'impact': {},
 'lastModifiedDate': '2018-04-16T09:58Z',
 'publishedDate': '2018-04-13T15:29Z'}

### above example has id CVE-2017-0356
### it has the link https://nvd.nist.gov/vuln/detail/CVE-2017-0357

# one can see that these are newly identified vulnerbalities that have not been put in the datafeeds 